### PR TITLE
Update custom TextField creation docs

### LINKED ISSUE #2343: Old documentation
**What you were expecting:**
Working material inputs

**What happened instead:**
Material inputs don't work - they do not show labels, errors, etc.

**Steps to reproduce:**
Create custom input following the docs: https://marmelab.com/react-admin/Inputs.html#writing-your-own-input-component

**Other information:**
Documentation about custom work with material inputs is old, when I changed code to this, everything works as expected:
```javascript
<TextField
      helperText={label}
      label={label}
      {...input}
      {...custom}
    />
```

**Environment**

* React-admin version: 2.3.0
* Last version that did not exhibit the issue (if applicable): -
* React version: 16.5.0
* Browser: -
* Stack trace (in case of a JS error): -

### PR DESCRIPTION
**Pull request by mikaoelitiana**
Fixes #2343

### CODE REVIEW

File: docs/Inputs.md [hunk 1]
```diff
@@ -1285,9 +1285,8 @@ import TextField from '@material-ui/core/TextField';
 import { Field } from 'redux-form';
 const renderTextField = ({ input, label, meta: { touched, error }, ...custom }) => (
     <TextField
-        hintText={label}
-        floatingLabelText={label}
-        errorText={touched && error}
+        helperText={label}
+        label={label}
```
Comment (djhi) [inline docs/Inputs.md]: I think we're missing the boolean prop `error` to display the helper text styled as an error

File: docs/Inputs.md [hunk 2]
```diff
@@ -1285,9 +1285,8 @@ import TextField from '@material-ui/core/TextField';
 import { Field } from 'redux-form';
 const renderTextField = ({ input, label, meta: { touched, error }, ...custom }) => (
     <TextField
-        hintText={label}
-        floatingLabelText={label}
-        errorText={touched && error}
+        helperText={label}
```
Comment (djhi) [inline docs/Inputs.md]: We should put the `label` here but the `error` (which is a string) if `touched` is `true`

**Comment by djhi** [comment]
Bump. Any news on this?

**Comment by fzaninotto** [comment]
No news for too long, closing.

**Pull request closed without merging**